src/zkevm_specs/evm/execution/gas.py

```diff
@@ -0,0 +1,22 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ..table import CallContextFieldTag, TxContextFieldTag
+
+
+def gas(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.GAS)
+
+    gas = instruction.stack_push()
+
+    instruction.constrain_equal(
+        gas,
+        instruction.curr.gas_left - Opcode.GAS.constant_gas_cost(),
+    )
```

Comment:
My last comment is about this comparison.  The gas value in the stack is RLC encoded whereas the gas_left is not, so the `gas` variable needs RLC "decoding".  I see that you already did this in the circuit implementation; could you also do it here? Thanks!

Reply:
ah, not sure how I missed that. I also did not encode the rw table pushed value to RLC encoded. Pushing a fix :)